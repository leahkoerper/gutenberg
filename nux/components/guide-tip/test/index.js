/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import { GuideTip } from '..';

describe( 'GuideTip', () => {
	it( 'should render correctly', () => {
		const wrapper = shallow(
			<GuideTip>
				It looks like you’re writing a letter. Would you like help?
			</GuideTip>
		);
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should not render when step does not equal currentStep', () => {
		const wrapper = shallow(
			<GuideTip step={ 1 } currentStep={ null }>
				It looks like you’re writing a letter. Would you like help?
			</GuideTip>
		);
		expect( wrapper.isEmptyRender() ).toBe( true );
	} );

	it( 'should call onAdvance when the next button is clicked', () => {
		const onAdvance = jest.fn();
		const wrapper = shallow(
			<GuideTip onAdvance={ onAdvance }>
				It looks like you’re writing a letter. Would you like help?
			</GuideTip>
		);
		wrapper.find( 'Button[children="Tell me more!"]' ).first().simulate( 'click' );
		expect( onAdvance ).toHaveBeenCalled();
	} );

	it( 'should call onDismiss when the close button is clicked', () => {
		const onDismiss = jest.fn();
		const wrapper = shallow(
			<GuideTip onDismiss={ onDismiss }>
				It looks like you’re writing a letter. Would you like help?
			</GuideTip>
		);
		wrapper.find( 'IconButton[label="Dismiss guide"]' ).first().simulate( 'click' );
		expect( onDismiss ).toHaveBeenCalled();
	} );
} );
