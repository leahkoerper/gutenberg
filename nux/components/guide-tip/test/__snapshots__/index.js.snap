// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuideTip should render correctly 1`] = `
<span>
  <GuideTipFill>
    <div
      aria-label="New User Guide"
      aria-modal="true"
      className="editor-guide-tip is-right"
      role="dialog"
      style={Object {}}
    >
      <div
        className="editor-guide-tip__content"
      >
        <p>
          It looks like youâ€™re writing a letter. Would you like help?
        </p>
        <p>
          <Button
            isLink={true}
          >
            Tell me more!
          </Button>
        </p>
        <IconButton
          className="editor-guide-tip__close"
          icon="no-alt"
          label="Dismiss guide"
        />
      </div>
    </div>
  </GuideTipFill>
</span>
`;
